agent/src/beerocks/slave/backhaul_manager/backhaul_manager_thread.cpp

```diff
@@ -2313,6 +2319,66 @@ bool backhaul_manager::handle_ap_capability_query(ieee1905_1::CmduMessageRx &cmd
     return send_cmdu_to_bus(cmdu_tx, src_mac, bridge_info.mac);
 }
 
+bool backhaul_manager::handle_ap_metrics_query(ieee1905_1::CmduMessageRx &cmdu_rx,
+                                               const std::string &src_mac)
+{
+    std::vector<sMacAddr> bssid;
+    const auto mid            = cmdu_rx.getMessageId();
+    auto ap_metrics_query_tlv = cmdu_rx.getClass<wfa_map::tlvApMetricQuery>();
+    if (!ap_metrics_query_tlv) {
+        LOG(ERROR) << "AP Metrics Query CMDU mid=" << mid << " does not have AP Metric Query TLV";
+        return false;
+    }
+    for (int bssid_idx = 0; bssid_idx < ap_metrics_query_tlv->bssid_list_length(); bssid_idx++) {
+        auto bssid_tupl = ap_metrics_query_tlv->bssid_list(bssid_idx);
+        if (!std::get<0>(bssid_tupl)) {
+            LOG(ERROR) << "Failed to get bssid " << bssid_idx << " from AP_METRICS_QUERY";
+            return false;
+        }
+        bssid.push_back(std::get<1>(bssid_tupl));
+        LOG(DEBUG) << "Received AP_METRICS_QUERY_MESSAGE, mid=" << std::hex << int(mid)
+                   << "  bssid " << std::get<1>(bssid_tupl);
+    }
+
+    auto cmdu_header = cmdu_tx.create(0, ieee1905_1::eMessageType::AP_METRICS_QUERY_MESSAGE);
+
+    if (!cmdu_header) {
+        LOG(ERROR) << "Failed building IEEE1905 AP_METRICS_QUERY_MESSAGE";
+        return false;
+    }
+    auto ap_metrics_response_tlv = cmdu_tx.addClass<wfa_map::tlvApMetricQuery>();
```

Comment:
Also I don't understand why you named it "response".
This is not a response, you _forward_ the message.
```suggestion
    auto ap_metrics_query_tlv = cmdu_tx.addClass<wfa_map::tlvApMetricQuery>();
```